sdk/workflow-tengo/src/workdir/index.lib.tengo

```diff
@@ -269,13 +303,41 @@ _fill := func() {
 		 * @return wd: reference - a reference to the filled workdir.
 		 */
 		build: func() {
-			emptyWorkdir := undefined
-			if feats.workdirLimits && smart.isReference(_allocation) {
-				emptyWorkdir = createV2(_allocation)
-			} else {
-				emptyWorkdir = create()
+			ll.assert(!is_undefined(_allocation) || !is_undefined(queue),
+				"workdir.builder: either allocation or queue must be provided")
+			ll.assert(is_undefined(_allocation) || is_undefined(queue),
+				"workdir.builder: allocation and queue must not be set at the same time")
+
+			if !is_undefined(_allocation) {
+				ll.assert(is_undefined(cpu) && is_undefined(ram),
+					"workdir.builder: cpu and ram cannot be set when allocation is provided")
 			}
 
+			allocation := _allocation
+			if is_undefined(allocation) {
+				quotaBuilder := limits.quotaBuilder()
+				if !is_undefined(cpu) {
+					quotaBuilder.cpu(cpu)
+				}
+				if !is_undefined(ram) {
+					quotaBuilder.ram(ram)
+				}
+
+				if limits.useBatch(queue) {
+					quotaBuilder.queue(execConstants.BATCH_QUEUE)
+				} else {
+					quotaBuilder.queue(queue)
+				}
+
+				storageSpaceRef := limits.storageSpaceRequestBuilder().
+					quota(quotaBuilder.build()).
+					build()
+
+				allocation = storageSpaceRef.allocation
```

Comment:
After brief discussion with @vgpopov , we decided that it is better to name all references with Ref suffix to be clear for the reader.

When you see `allocation` you can think of real allocation resource to be there. When you see `allocationRef` it is clear for you that there might be unresolved reference under the hood.

Reply:
Ok, good point. Though there are many places where similar naming would be useful, this is one out of hundreds of similar places.